src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -2685,5 +2673,890 @@ public static int M()
                 Assert.Equal(expected, actual);
             });
         }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        [InlineData("string", "11")]
+        [InlineData("object", "18")]
+        [InlineData("C", "18")]
+        public void ReadOnlySpanFromArrayOfConstants_Null(string type, string typeCode)
+        {
+            var src = $$"""
+var values = C.M();
+System.Console.Write($"{values.Length} {values[0] is null} {values[1] is null} {values[2] is null}");
+
+public class C
+{
+    public static System.ReadOnlySpan<{{type}}> M()
+        => new {{type}}[] { null, null, null };
+
+    public static System.ReadOnlySpan<{{type}}> M2()
+        => new {{type}}[] { null, null, null };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            var verifier = CompileAndVerify(compilation, expectedOutput: "3 True True True", verify: Verification.Skipped);
+            var expectedIL = $$"""
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  IL_0000:  ldsfld     "{{type}}[] <PrivateImplementationDetails>.709E80C88487A2411E1EE4DFB9F22A861492D20C4765150C0C794ABD70F8147C_B{{typeCode}}"
```

Comment:
>{{type}} [](http://example.com/codeflow?start=24&length=8)

Looking at this scenario again, I do think there's a problem. The type of the `PrivateImplementationDetails` field should be `object[]` for all references. But it seems order-dependent. I'll take a look